linera-chain/src/chain.rs

```diff
@@ -736,8 +772,12 @@ where
         Self::sub_assign_fees(balance, policy.block_price(), ChainExecutionContext::Block)?;
 
         // Recompute the state hash.
-        let state_hash = self.execution_state.crypto_hash().await?;
-        self.execution_state_hash.set(Some(state_hash));
+        let state_hash = {
+            let _hash_latency = STATE_HASH_COMPUTATION_LATENCY.measure_latency();
+            let state_hash = self.execution_state.crypto_hash().await?;
+            self.execution_state_hash.set(Some(state_hash));
```

Comment:
I would have put the setting of the state hash outside of that bracketed statement since it is not really part of the hash computation. Plus the code would be nicer.

Reply:
Good point, fixed. Thanks!